code/game/objects/items/weapons/melee/energy_melee_weapons.dm

```diff
@@ -26,6 +26,11 @@
 	. = ..()
 	force_off = initial(force) //We want to check this only when initializing, not when swapping, so sharpening works.
 	throwforce_off = initial(throwforce)
+	RegisterSignal(src, COMSIG_ITEM_SHARPEN_ACT, PROC_REF(try_sharpen))
+
+/obj/item/melee/energy/Destroy()
+	. = ..()
+	UnregisterSignal(src, COMSIG_ITEM_SHARPEN_ACT)
```

Comment:
```suggestion
	UnregisterSignal(src, COMSIG_ITEM_SHARPEN_ACT)
	return ..()
```
you should do cleanup in Destroy() first and then return parent

Reply:
Will do!
(I also forgot the Signal handler on the proc lmfao)